### PR TITLE
[docs] add app archetype flow docs

### PR DESCRIPTION
**Pull request by didi0613**

### CODE REVIEW

File: docs/chapter1/intermediate/app-archetype/flow.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+[libs]
+include the specified library definitions when type checking your code.
+```
+
+Checkout [here](https://flow.org/en/docs/config/) for more information.
+
+##### - prepare the code
+
+`Flow` will onlu be checked when user has flow code. To mark the flow files, you can place `// @flow` or `/* @flow */` before any code in Javascript files.
```
Comment (datvong-wm) [inline docs/chapter1/intermediate/app-archetype/flow.md]: onlu

File: docs/chapter1/intermediate/app-archetype/flow.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+This flag is in the form of a normal JavaScript comment annotated with @flow. The Flow background process gathers all the files with this flag and uses the type information available from all of these files to ensure consistency and error free programming.
+
+##### - check the code
+
+If you wants to run the Flow Background Process, you can start the process that allows Flow to check your code incrementally and with great speed by `flow status`.
+
+To stop the background process, run flow stop.
+
+You can also type flow to accomplish the same effect as status is the default flag to the flow binary. At any point that you want to check for errors, just run `flow`.
```
Comment (datvong-wm) [inline docs/chapter1/intermediate/app-archetype/flow.md]: include some links to flow documentation

Comment (didi0613) [reply docs/chapter1/intermediate/app-archetype/flow.md]: @datvong-wm updated

File: docs/chapter1/intermediate/app-archetype/flow-faq.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Flow type checker frequent ask
+
+Below are the issues that frequently being ask, you can update them if you see any similar issue:
```
[the remark this answers is not in the document]
Comment (didi0613) [reply docs/chapter1/intermediate/app-archetype/flow-faq.md]: These are the issues i met when i was adding flow to example app.

**Pull request merged**